### PR TITLE
Share Space: Add Invite Someone Support

### PR DESCRIPTION
**Pull request by Kokobe**
Adds Invite Someone textfields behind a feature flag. 
- Reorganizes `ShareSpaceUI` code into `ShareSpaceSheet`, `InviteUserUI`, `ShareSpaceLinkUI`.
- Rename `OnboardingTextField` to `ClearFocusOnDismissTextField`
- Incorporates new designs to `InviteUserUI`. 
- Adds new spaces.graphql and adds `AddSpaceSoloACLsMutation` mutation functionality.

### Pointers to help with review
* [Project Proposal](https://paper.dropbox.com/doc/Project-Proposal-Invite-User-to-Space--BoTIEQTpdtJzx4nBppaYQB6GAg-tnjXvk1hYuEZNujCsYR6u)
* [Figma](https://www.figma.com/file/D3pbLY6ZP5YtCjTYqsGOKO/Android-Design-System?node-id=2070%3A33760)
* Link to where the change is discussed: https://app.clickup.com/36034046/v/l/12bnfy-14420?pr=54170898

### How was this tested?
- [x] I tested this manually
- [x] I added automated tests
- [x] Existing automated tests cover some of my changes
- [ ] Existing automated tests cover all of my changes

### Screenshots and screen recordings

Pressing `ShareLink` Buttons will automatically change the Space to `public`. 

https://user-images.githubusercontent.com/20916043/191618037-64e555c6-69d4-4909-8f15-5ceab9823c0e.mp4

Screen Shot 2022-09-21 at 3 09 38 PM

Screen Shot 2022-09-21 at 3 09 43 PM

### CODE REVIEW

File: app/src/main/graphql/com/neeva/app/spaces.graphql [hunk 1]
```diff
@@ -176,3 +176,10 @@ mutation LeaveSpace($input: LeaveSpaceInput!) {
 mutation DeleteSpace($input: DeleteSpaceInput!) {
   deleteSpace(input: $input)
 }
+
+mutation AddSpaceSoloACLs($input: AddSpaceSoloACLsInput!) {
+  addSpaceSoloACLs(input: $input) {
```
Comment (dan-alcantara-neeva) [inline app/src/main/graphql/com/neeva/app/spaces.graphql:181]: Indentation is off here

File: app/src/main/java/com/neeva/app/appnav/AppNavModelImpl.kt [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 package com.neeva.app.appnav
 
+import ShareSpaceSheet
```
Comment (dan-alcantara-neeva) [inline app/src/main/java/com/neeva/app/appnav/AppNavModelImpl.kt:7]: Is this import actually necessary?

**Comment by dan-alcantara-neeva** [comment]
What does this look like if the font is 2x the normal size?

**Comment by Kokobe** [comment]
> What does this look like if the font is 2x the normal size?

Screen Shot 2022-09-21 at 1 39 48 PM

Are you talking about the textfields or just all of the Text in the screenshot in general?

File: app/src/main/java/com/neeva/app/spaces/share/ShareSpaceLinkUI.kt [hunk 3]
```diff
@@ -21,151 +19,59 @@ import androidx.compose.foundation.layout.fillMaxWidth
...
-                }
-            }
-
-            SocialShareRow(spaceURL = spaceURL, onCopyLink = onCopyLink, onMore = onMore)
-        }
     }
 }
 
 @Composable
 fun SocialShareRow(
```
Comment (dan-alcantara-neeva) [inline app/src/main/java/com/neeva/app/spaces/share/ShareSpaceLinkUI.kt:72]: What happens to the buttons if the screen is small and the font is big?

File: app/src/main/java/com/neeva/app/spaces/share/InviteUserUI.kt [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+                    }
+                )
+            }
+        }
+    }
+}
+
+private fun getACLText(aclLevel: SpaceACLLevel): String {
+    return when (aclLevel) {
+        SpaceACLLevel.View -> "Can view"
```
Comment (dan-alcantara-neeva) [inline app/src/main/java/com/neeva/app/spaces/share/InviteUserUI.kt:182]: Are you hard-coding user-visible strings?  You should be putting these and all the other ones in the `strings.xml`.

File: app/src/main/java/com/neeva/app/spaces/share/InviteUserUI.kt [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+        expanded = expanded.value,
+        onExpandedChange = {
+            expanded.value = !expanded.value
+        }
+    ) {
+        OutlinedTextField(
+            value = getACLText(selectedACLLevel.value),
+            onValueChange = {},
+            readOnly = true,
+            label = { Text(text = "Permissions") },
```
Comment (dan-alcantara-neeva) [inline app/src/main/java/com/neeva/app/spaces/share/InviteUserUI.kt:156]: strings.xml

**Review by dan-alcantara-neeva — CHANGES_REQUESTED** [review]
Initial comments

**Comment by dan-alcantara-neeva** [comment]
> > What does this look like if the font is 2x the normal size?
> 
> Screen Shot 2022-09-21 at 1 39 48 PM
> 
> Are you talking about the textfields or just all of the Text in the screenshot in general?

Everything in that screenshot.  Want to see what the social buttons do.

**Comment by Kokobe** [comment]
Oh I misread your comment. I'll add those to the description and re-request review

File: app/src/main/java/com/neeva/app/appnav/AppNavModelImpl.kt (hunk 2 above)
Comment (Kokobe) [reply app/src/main/java/com/neeva/app/appnav/AppNavModelImpl.kt:7]: Yeah, I'm having trouble making this import say its from com.neeva.app.spaces.share.ShareSpaceSheet

File: app/src/main/java/com/neeva/app/spaces/share/ShareSpaceLinkUI.kt (hunk 3 above)
Comment (Kokobe) [reply app/src/main/java/com/neeva/app/spaces/share/ShareSpaceLinkUI.kt:72]: Checking with @Ria-yu , I can make it wrap around like what you did with `CookieCutterPopover`

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -326,14 +326,44 @@
...
+    <string name="share_space_can_comment_permission">Can comment</string>
+
+    <!-- [Spaces] When sharing a Space to another user, this text is used for the dropdown menu item "Can Edit" permission option. Granting a given user "Can Edit" permissions means the given user can view, comment, and edit the Space. This is not the same as owning the Space. -->
+    <string name="share_space_can_edit_permission">Can edit</string>
+
+    <!-- [Spaces] Title of Dropdown Menu in the given Space that allows user to choose what permissions they will grant the user they invite. -->
+    <string name="share_space_permissions">Permissions</string>
+
+    <!-- [Spaces] A snackbar popup message confirming that the invitation filled out by the Share Space bottom sheet has been sent. -->
+    <string name="share_space_confirmation_message">Invitation sent.</string>
```
Comment (dan-alcantara-neeva) [inline app/src/main/res/values/strings.xml:345]: Do we normally put periods for popups?  Can you check?

File: app/src/main/java/com/neeva/app/spaces/share/ShareSpaceLinkUI.kt [hunk 7]
```diff
@@ -21,151 +19,59 @@ import androidx.compose.foundation.layout.fillMaxWidth
...
             primaryLabel = stringResource(id = R.string.enable_link),
             secondaryLabel = stringResource(id = R.string.enable_link_subtitle),
-            isChecked = isSpacePublic,
-            onCheckedChange = onTogglePublic
+            isChecked = params.isSpacePublic,
+            onCheckedChange = params.onTogglePublic
+        )
+        SocialShareRow(
+            spaceURL = params.spaceURL,
+            onClickSocialShareButton = { params.onTogglePublic(true) },
```
Comment (dan-alcantara-neeva) [inline app/src/main/java/com/neeva/app/spaces/share/ShareSpaceLinkUI.kt:64]: Name the argument in `onTogglePublic`

File: app/src/main/java/com/neeva/app/appnav/AppNavModelImpl.kt (hunk 2 above)
Comment (dan-alcantara-neeva) [reply app/src/main/java/com/neeva/app/appnav/AppNavModelImpl.kt:7]: Eh... I don't understand why your import is the only one in the entire repo that doesn't behave right.  Can you look into this, or say what you've tried?

File: app/src/main/res/values/strings.xml (hunk 6 above)
Comment (Kokobe) [reply app/src/main/res/values/strings.xml:345]: It's inconsistent, Spaces does not use periods. But Settings and HistoryUI does. Good catch! Will ask design and update this thread. And also file an issue about this

Comment (Kokobe) [reply app/src/main/res/values/strings.xml:345]: From @yozuysal on Slack:
```
Periods

Avoid using periods in solitary sentences in these elements:

Labels
Hover text
Bulleted lists
Dialog body text
Periods should be used on:

Multiple sentences
Any sentence followed by a link
```

https://material.io/archive/guidelines/style/writing.html#writing-capitalization-punctuation

Comment (Kokobe) [reply app/src/main/res/values/strings.xml:345]: Going to do a quick refactor in this PR. Will make it 1 clean commit

File: app/src/main/java/com/neeva/app/appnav/AppNavModelImpl.kt (hunk 2 above)
Comment (dan-alcantara-neeva) [reply app/src/main/java/com/neeva/app/appnav/AppNavModelImpl.kt:7]: @Kokobe

Comment (Kokobe) [reply app/src/main/java/com/neeva/app/appnav/AppNavModelImpl.kt:7]: I found out the problem: I forgot to add the package at the top of ShareSpaceSheet

**Review by dan-alcantara-neeva — APPROVED** [review]

**Pull request merged**